dataloader.py

```diff
@@ -0,0 +1,113 @@
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Tuple
+
+import cv2
+import torch
+from skimage import io
+from torch import Tensor
+from torch.utils.data import Dataset
+from torchvision.transforms import RandomResizedCrop
+
+from constants import NO_VAL, TEST_DIR, TRAIN_DIR, VAL_DIR
+
+# Make a hashmap for both datasets
+
+
+class ChestXrayDataset(Dataset):
+    def __init__(
+        self,
+        phase: str,
```

Comment:
Since `phase` here can only actually be either `"train"` or `"val"`, it would be better to annotate this exactly, either with 

```suggestion
        phase: Literal["train", "val"],
```

or by defining a `Phase` enum, and using that:

```suggestion
from enum import Enum

class Phase(Enum):
    Train = "train"
    Val = "val"
    Test = "test"
    Predict = "pred"
    
class ChestXrayDataset(Dataset):
    def __init__(
        self,
        phase: Phase,
```

Reply:
Implemented `Enum`